.github/workflows/golangci-lint.yml

```diff
@@ -17,10 +17,21 @@ jobs:
       - name: Checkout EigenDA
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #4.2.2
 
+      # Fetch git history of the base branch so golangci-lint can analyze only the diff
+      # between the feature branch and base branch (via --new-from-rev flag)
+      - name: Fetch base branch history
+        if: github.event_name == 'pull_request'
+        run: bash .github/scripts/fetch-base-incrementally.sh "${{ github.event.pull_request.base.ref }}"
+
       - uses: jdx/mise-action@v2
         with:
           version: ${{ env.MISE_VERSION }}
           experimental: true
       - run: go version
-      - run: make lint
+      
+      - name: Run linter
+        run: |
+          # Use PR base branch if available, otherwise the command line flag is ignored and master is used
+          golangci-lint run --new-from-rev=origin/${{ github.event.pull_request.base.ref || 'master' }}
```

Comment:
hmm so by default itll lint against master branch? But doing this will lint against the PR target branch instead?

For ultra simplicity, I would suggest KISS and only run when target branch is master? So that we don't need this complex fetch-base-incrementally script that I dont understand (didnt take the time to read it tbh)

Reply:
Otherwise I would add this comment above the Makefile `lint` target, something like:
"this will lint by comparing against master. To comapre against another branch, run `golangci-lint run --new-from-rev=...`"

Reply:
see https://github.com/Layr-Labs/eigenda/pull/1817#discussion_r2245581092

TLDR; keeping the bit that runs this against the target, but ugly script is gone